src/utils.js

```diff
@@ -157,17 +157,50 @@ export function parseGPTSingleSizeArray(singleSize) {
   }
 };
 
-exports.getTopWindowLocation = function () {
-  let location;
+export function getTopWindowLocation() {
```

Comment:
Several modules rely on this function, changing it may introduce subtle differences. Any reason to not keep the existing `getTopWindowLocation` as is, and create a new function for this case?

Reply:
The old function `getTopWindowLocation` either returns window.top.location or returns window.location.

The new `getTopWindowLocation` returns window.location, window.top.location, window.document.referrer, window.document.location.ancestorOrigins[0], or an object that recreates the values within window.location

The new version attempts to get the same info, just from additional objects.

I checked the modules, and I think this new version should fit all the current use cases.

Though, I am not opposed to creating a new function if you’re worried about it.

Reply:
Ok, thanks for the explanation. Changes to core require a second review, assigning that now